scene/resources/primitive_meshes.cpp

```diff
@@ -789,7 +789,7 @@ void CylinderMesh::create_mesh_array(Array &p_arr, float top_radius, float botto
 
 			Vector3 p = Vector3(x * radius, y, z * radius);
 			points.push_back(p);
-			normals.push_back(Vector3(x, 0.0, z));
+			normals.push_back(Vector3(x, (bottom_radius - top_radius)/height, z).normalized());
```

Comment:
This by component is constant so it could be computed only once outside the for loop.
Style wise, you should also have spaces around `/` (run clang-format locally to make sure to us the right style).